chromium_src/components/permissions/android/permission_prompt/permission_dialog_delegate.cc

```diff
@@ -62,20 +72,57 @@ void ApplyLifetimeToPermissionRequests(
   }
 }
 
-void Java_PermissionDialogController_createDialog_BraveImpl(
-    JNIEnv* env,
-    const base::android::JavaRef<jobject>& delegate) {
-  SetLifetimeOptions(delegate);
-  Java_PermissionDialogController_createDialog(env, delegate);
+#if BUILDFLAG(ENABLE_WIDEVINE)
```

Comment:
It seems weird to me that we're doing widevine specific stuff in here. @goodov is this the right place for this? There's the obvious layering violation issue that @simonhong pointed out. If we don't have this problem on desktop, it doesn't seem like we should have it on android either

Reply:
we should generalize "dont ask again" permission option and make it a part of `PermissionRequest`.

Reply:
I have moved `dont_ask_again_` up to `PermissionRequest` to be a generic property. The only widevine specific function left is `Java_BravePermissionDialogDelegate_setIsWidevinePermissionRequest` which is needed because Widevine doesn't have a special Content type which is needed for Java to know what kind of request it is getting.